### PR TITLE
158: Merge ID3 into Master

### LINKED ISSUE #158: Merge ID3 branch into master for the ID3 submission
Merge in the work from ID3 into the master branch for tagging

### PR DESCRIPTION
**Pull request by matthew-buglass**
Closes #158

# What was the issue
This is the collection of our work for ID3 that is to be tested and validated before being merged into master.

# What was done and why
Our goal for this sprint was to get a full happy path through the application and to build out and refactor some back-end processes. We didn't get in the happy path, due to some tricky merge conflicts, but it should be merged in early next sprint.

# How it was tested
To verify the latest release, the web application was tested in both local and firebase environments. As well as verified on Meta Quest devices. To verify, each gherkin file was walked through to verify all old and new cases still pass as expected, with deeper general use cases for the “feel” of the application. Thus running along side all passing unit cases gives us confidence that it is ready for release, and further testing to be completed.

### CODE REVIEW

**Review by qpb948 — APPROVED** [review]
We need to figure out why the case agnostic thingy happened again with Jesse. We were able to catch it this time thanks to the linter I believe. But why did it happen in the first place.

**Comment by Subzeero** [comment]
Did some investigating into this. It looks like it was just missed during a merge: 
> https://github.com/UniversityOfSaskatchewanCMPT371/term-project-2024-team-2/pull/156/commits/197a96d4ee7fb5ac2a80e009e6b0a58eb41dcf3e#diff-f1a8a431074765d5b1b71567b88b594ca80283193bac15daebdf08dc78546eee

`singleAxis.tsx` was modified (instead of deleted) and `SingleAxis.tsx` was created. If this merge was handled locally on MacOS or Windows, it could've been impossible to tell the difference.

**Comment by matthew-buglass** [comment]
> > We need to figure out why the case agnostic thingy happened again with Jesse. We were able to catch it this time thanks to the linter I believe. But why did it happen in the first place.
> 
> Did some investigating into this. It looks like it was just missed during a merge:
> 
> > [197a96d#diff-f1a8a431074765d5b1b71567b88b594ca80283193bac15daebdf08dc78546eee](https://github.com/UniversityOfSaskatchewanCMPT371/term-project-2024-team-2/commit/197a96d4ee7fb5ac2a80e009e6b0a58eb41dcf3e#diff-f1a8a431074765d5b1b71567b88b594ca80283193bac15daebdf08dc78546eee)
> 
> `singleAxis.tsx` was modified (instead of deleted) and `SingleAxis.tsx` was created. If this merge was handled locally on MacOS or Windows, it could've been impossible to tell the difference as they would've shown as the same file.

If it's not blocking, then I'd say we fix it in the next ID.

**Review by MitchWag01 — APPROVED** [review]
Everything looks good

File: my-webxr-app/src/LogConfig.tsx
Comment (LongQuanNguyen) [inline my-webxr-app/src/LogConfig.tsx:1]: Should we delete this file and its tests before merge in?

**Review by LongQuanNguyen — APPROVED** [review]
I am still a bit skeptical on the nodePolyfills plug-in that we use to get assert working, I don't see any problem now but will keep an eye out for it. Everything else looks good.

File: my-webxr-app/src/LogConfig.tsx
Comment (Subzeero) [reply my-webxr-app/src/LogConfig.tsx:1]: At this point in the release cycle, this file has no harm staying in. When I refactor the logging setup in ID4, this file will be removed.

**Pull request merged**